.github/workflows/test_on_push.yml

```diff
@@ -299,6 +299,10 @@ jobs:
           pip install --upgrade pip wheel setuptools nox
           pip install -e .[all,docs]
 
+      - name: Test pybamm_install_odes on MacOS (for only this PR)
+        if: matrix.os == 'macos-latest'
+        run: pybamm_install_odes
```

Comment:
Note: this step won't run at all currently because the job that it is in runs only on Ubuntu. Adding a parallel job (in a matrix for GNU/Linux and macOS) that needs the style tests would be better because we would want to use a fresh runner

Reply:
My bad, I added it to wrong place thanks.

Reply:
It is still a step in another job, could you create a different job with steps that check out the repository but don't run `pybamm-requires` beforehand?